livre/data/DFSTraversal.cpp

```diff
@@ -17,53 +17,30 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include <livre/core/data/DataSource.h>
-#include <livre/core/visitor/DFSTraversal.h>
-#include <livre/core/visitor/NodeVisitor.h>
-#include <livre/core/visitor/VisitState.h>
-#include <livre/core/visitor/VisitState.h>
+#include <livre/data/DFSTraversal.h>
+#include <livre/data/DataSource.h>
+#include <livre/data/NodeVisitor.h>
 
 namespace livre
 {
 struct DFSTraversal::Impl
 {
 public:
-    bool traverse(const NodeId& nodeId, const uint32_t depth,
+    void traverse(const NodeId& nodeId, const uint32_t depth,
                   livre::NodeVisitor& visitor)
     {
-        if (depth == 0 || _state.getBreakTraversal())
-            return false;
+        assert(depth > 0);
 
-        _state = VisitState();
+        if (!visitor.visit(nodeId))
+            return;
 
-        if (_state.getBreakTraversal())
-            return true;
-
-        visitor.visit(nodeId, _state);
-
-        if (_state.getBreakTraversal() || !_state.getVisitChild())
-        {
-            _state.setVisitChild(true);
-            return false;
-        }
+        if (depth == 1)
+            return;
 
         const NodeIds& nodeIds = nodeId.getChildren();
         for (const NodeId& childNodeId : nodeIds)
-        {
             traverse(childNodeId, depth - 1, visitor);
```

Comment:
Don't see why you could stop traversing the rest of the siblings if one stops traversing. What do you want to have changed?